### PR TITLE
feat: implement product-service

### PR DESCRIPTION
**Pull request by noway36**
[Frontend PR](https://github.com/noway36/shop-react-redux-cloudfront/pull/2) (Lots of formatting, sry for that 🙂)
[Product-service API](https://ynbaj9n6f2.execute-api.eu-west-1.amazonaws.com/dev/products)

1) Frontend application is integrated with product service (/products API) and products from product-service are represented on Frontend. AND POINT1 and POINT2 are done.
2) Async/await is used in lambda functions
3) ES6 modules are used for product-service implementation
4) Webpack is configured for product-service
5) SWAGGER documentation is created for product-service (openapi.yml - npm run docs:update will automatically update it, also you can see the docs [here](https://d1mj375gxygqlt.cloudfront.net/api-docs)) - initially was implemented on ui side, so I decided to leave /api-docs url
6) Lambda handlers are covered by basic UNIT tests
7) Lambda handlers code is written not in 1 single module (file) and separated in codebase.
8) Main error scenarious are handled by API ("Product not found" error, try catch blocks are used in lambda handlers).

In case of any questions please contact Aliaksandr Dren in Teams.

### CODE REVIEW

File: product-service/openapi.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        '200':
+          description: all available products
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ProductItemsArray'
+        '500':
+          description: all available products
+          content: {}
+  '/product/{productId}':
```
Comment (DrDiman) [inline product-service/openapi.yml]: I guess that it should be as /product**S**/{productId}

File: product-service/src/functions/get-product/handler.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+    if (productId != null) {
+        const product = PRODUCT_BY_ID_MAP.get(productId);
+
+        if (product != null) {
+            return formatJSONResponse({
+                product,
+            });
+        }
+    }
```
Comment (DrDiman) [inline product-service/src/functions/get-product/handler.ts]: u could use try catch block 

e.g. emulate service for fetching products as:

const getProducts = () => Promise.resolve(YOUR_MOCK)

and then here wrap it in tryCatch

try {
  const products = await getProducts()
} catch {}

File: product-service/src/libs/apiGateway.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+const headers = {
+    'Access-Control-Allow-Origin': '*',
+};
+
+export const formatJSONResponse = (response: Record<string, unknown>): IResponse => {
+    return {
+        statusCode: HttpCode.Ok,
+        body: JSON.stringify(response),
+        headers: {
+            ...headers,
```
Comment (DrDiman) [inline product-service/src/libs/apiGateway.ts]: any reasons to spread and copy content? how about just link?)

File: product-service/package.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        "build": "sls package",
+        "docs:update": "sls openapi generate"
+    },
+    "engines": {
+        "node": ">=14.15.0"
+    },
+    "dependencies": {
+        "@middy/core": "^1.5.2",
+        "@middy/http-json-body-parser": "^1.5.2",
+        "jest": "^27.0.6",
```
Comment (DrDiman) [inline product-service/package.json]: devDependency for sure

File: product-service/package.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        "source-map-support": "^0.5.19"
```
Comment (DrDiman) [inline product-service/package.json:19]: devDependency  as well?

**Review by DrDiman — COMMENTED** [review]
looks good. 11 points go tooo @noway36!))

**Comment by noway36** [comment]
@DrDiman thanks for the review, everything is fixed, except for the last one. It seems that I should leave `source-map-support` package as it is.
>Serverless: ERROR: Runtime dependency 'source-map-support' found in devDependencies. Move it to dependencies or use forceExclude to explicitly exclude it.

**Pull request merged**